cosmos/airflow/graph.py

```diff
@@ -66,13 +66,18 @@ def create_test_task_metadata(
     :param task_args: Arguments to be used to instantiate an Airflow Task
     :param on_warning_callback: A callback function called on warnings with additional Context variables “test_names”
     and “test_results” of type List.
-    :param model_name: If the test relates to a specific model, the name of the model it relates to
+    :param node: If the test relates to a specific node, the node reference
     :returns: The metadata necessary to instantiate the source dbt node as an Airflow task.
     """
     task_args = dict(task_args)
     task_args["on_warning_callback"] = on_warning_callback
-    if model_name is not None:
-        task_args["models"] = model_name
+    if node is not None:
+        if node.resource_type == DbtResourceType.MODEL:
+            task_args["models"] = node.name
+        elif node.resource_type == DbtResourceType.SOURCE:
+            task_args["select"] = f"source:{node.unique_id[len('source.'):]}"
```

Comment:
do we have a test case that's covering this part?

Reply:
@harels, we have a unit that covers the path formation for all these variations:
`test_create_test_task_metadata,` and I manually played with dbt 1.6 to confirm all these variations work.

ATM, we don't have an integration test for `DbtResourceType.SOURCE` since our only example DAG that validates this type of node overrides any default behavior. For us to have an end-to-end version of the code going through this, Cosmos would need a default behavior for SOURCE nodes, something we don't currently have.

@harels @jlaneve Should we add a ticket to define a default behavior for Source nodes? What do you think that should look like?  Should it check for freshness?